src/read.c

```diff
@@ -161,6 +156,10 @@ static UInt findValueInNams(Obj nams, const Char * val, UInt start, UInt end)
 */
 
 
+// HACK
+#define s (&rs->s)
```

Comment:
This HACK is part of the second commit (which is meant to be dropped before merging this), and only is here for review purposes, to reduce the number of meaningless diffs: where before we directly accessed a `ScannerState` named `s`, we now instead need to get the `ScannerState` from inside the `ReaderState` called `rs`. To hide this, we use this `#define`.